pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -539,12 +539,27 @@ func (rc *reconciler) updateStates(volumesNeedUpdate map[v1.UniqueVolumeName]*re
 			continue
 		}
 		if volume.pluginIsAttachable {
-			err = rc.actualStateOfWorld.MarkDeviceAsMounted(volume.volumeName)
+			// Get attacher, if possible
+			attachableVolumePlugin, _ :=
+				rc.volumePluginMgr.FindAttachablePluginBySpec(volume.volumeSpec)
+			var volumeAttacher volumepkg.Attacher
+			if attachableVolumePlugin != nil {
+				volumeAttacher, _ = attachableVolumePlugin.NewAttacher()
+			}
+			if volumeAttacher != nil {
+				volume.devicePath, err = volumeAttacher.WaitForAttach(
+					volume.volumeSpec, "" /* devicePath */, volume.pod, rc.waitForAttachTimeout)
```

Comment:
if the devicePath is empty here, AWS WaitForAttach will fail. Is it something we need worry about?

Reply:
@jingxu97 
In order to cover WaitForAttach() failure case, can we call MarkDeviceAsMounted() twice at line 542 and line 557?

- At the call on L542, the we pass "" for devicePath then the volume is marked as mounted but the device path isn't updated.
- At the call on L557, we pass actual devicePath if we succeed WaitForAttach() then volume is marked as mounted and also devicePath is updated.

Reply:
I am not sure it will work. Seem like as aws.WaitForAttach() will always fail if devicePath is empty, and it returns empty string again.

Reply:
@jingxu97 
Based on conversation on slack, we should check length of volume.devicePath beforehand.
- just run MarkDeviceAsMounted() if length of volume.devicePath > 0
- otherwise, update devicePath via WaitForAttach() then run MarkDeviceAsMounted() with returned devicePath from WaitForAttach()

Reply:
actually I think the previous implementation is ok. If volume type if AWS, the devicePath retrieved from node status should not be empty. Only for FC or Iscsi might have empty device path. sorry for the confusion.